### ISSUE TITLE
Unable to create "hello" phoenix application on macOS, can't fetch tailwind, unknown Certificate Authority

### ISSUE THREAD

**Issue by SeanShubin** [phoenixframework/phoenix#5339]
### Environment

* Elixir version (elixir -v):
    * Erlang/OTP 25 [erts-13.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit:ns] [dtrace]
    * Elixir 1.14.3 (compiled with Erlang/OTP 25)

* Phoenix version (mix deps):
```
* castore 0.1.22 (Hex package) (mix)
  locked at 0.1.22 (castore) c17576df
  ok
* connection 1.1.0 (Hex package) (mix)
  locked at 1.1.0 (connection) 722c1eb0
  ok
* cowboy 2.9.0 (Hex package) (rebar3)
  locked at 2.9.0 (cowboy) 2c729f93
  ok
* cowboy_telemetry 0.4.0 (Hex package) (rebar3)
  locked at 0.4.0 (cowboy_telemetry) 7d98bac1
  ok
* cowlib 2.11.0 (Hex package) (rebar3)
  locked at 2.11.0 (cowlib) 2b3e9da0
  ok
* db_connection 2.4.3 (Hex package) (mix)
  locked at 2.4.3 (db_connection) c127c15b
  ok
* decimal 2.0.0 (Hex package) (mix)
  locked at 2.0.0 (decimal) 34666e9c
  ok
* ecto 3.9.4 (Hex package) (mix)
  locked at 3.9.4 (ecto) de5f988c
  ok
* ecto_sql 3.9.2 (Hex package) (mix)
  locked at 3.9.2 (ecto_sql) 1eb5eeb4
  ok
* esbuild 0.6.1 (Hex package) (mix)
  locked at 0.6.1 (esbuild) 569f7409
  ok
* expo 0.4.0 (Hex package) (mix)
  locked at 0.4.0 (expo) a8ed1683
  ok
* file_system 0.2.10 (Hex package) (mix)
  locked at 0.2.10 (file_system) 41195edb
  ok
* finch 0.14.0 (Hex package) (mix)
  locked at 0.14.0 (finch) 5459acaf
  ok
* gettext 0.22.1 (Hex package) (mix)
  locked at 0.22.1 (gettext) ad105b8d
  ok
* hpax 0.1.2 (Hex package) (mix)
  locked at 0.1.2 (hpax) 2c87843d
  ok
* jason 1.4.0 (Hex package) (mix)
  locked at 1.4.0 (jason) 79a37910
  ok
* mime 2.0.3 (Hex package) (mix)
  locked at 2.0.3 (mime) 27a30bf0
  ok
* mint 1.5.1 (Hex package) (mix)
  locked at 1.5.1 (mint) 4a63e1e7
  ok
* nimble_options 0.5.2 (Hex package) (mix)
  locked at 0.5.2 (nimble_options) 4da7f904
  ok
* nimble_pool 0.2.6 (Hex package) (mix)
  locked at 0.2.6 (nimble_pool) 1c715055
  ok
* phoenix 1.7.1 (Hex package) (mix)
  locked at 1.7.1 (phoenix) ea9d4a85
  ok
* phoenix_ecto 4.4.0 (Hex package) (mix)
  locked at 4.4.0 (phoenix_ecto) 09864e55
  ok
* phoenix_html 3.3.1 (Hex package) (mix)
  locked at 3.3.1 (phoenix_html) bed1906e
  ok
* phoenix_live_dashboard 0.7.2 (Hex package) (mix)
  locked at 0.7.2 (phoenix_live_dashboard) 0e5fdf06
  ok
* phoenix_live_reload 1.4.1 (Hex package) (mix)
  locked at 1.4.1 (phoenix_live_reload) 9bffb834
  ok
* phoenix_live_view 0.18.17 (Hex package) (mix)
  locked at 0.18.17 (phoenix_live_view) f4b5710e
  ok
* phoenix_pubsub 2.1.1 (Hex package) (mix)
  locked at 2.1.1 (phoenix_pubsub) 81367c6d
  ok
* phoenix_template 1.0.1 (Hex package) (mix)
  locked at 1.0.1 (phoenix_template) 157dc078
  ok
* plug 1.14.0 (Hex package) (mix)
  locked at 1.14.0 (plug) bf020432
  ok
* plug_cowboy 2.6.0 (Hex package) (mix)
  locked at 2.6.0 (plug_cowboy) 073cf20b
  ok
* plug_crypto 1.2.3 (Hex package) (mix)
  locked at 1.2.3 (plug_crypto) b5672099
  ok
* postgrex 0.16.5 (Hex package) (mix)
  locked at 0.16.5 (postgrex) edead639
  ok
* ranch 1.8.0 (Hex package) (rebar3)
  locked at 1.8.0 (ranch) 49fbcfd3
  ok
* swoosh 1.9.1 (Hex package) (mix)
  locked at 1.9.1 (swoosh) 76dffff3
  ok
* tailwind 0.1.10 (Hex package) (mix)
  locked at 0.1.10 (tailwind) e0fc474d
  ok
* telemetry 1.2.1 (Hex package) (rebar3)
  locked at 1.2.1 (telemetry) dad9ce9d
  ok
* telemetry_metrics 0.6.1 (Hex package) (mix)
  locked at 0.6.1 (telemetry_metrics) 7be9e087
  ok
* telemetry_poller 1.0.0 (Hex package) (rebar3)
  locked at 1.0.0 (telemetry_poller) b3a24eaf
  ok
* websock 0.5.0 (Hex package) (mix)
  locked at 0.5.0 (websock) b51ac706
  ok
* websock_adapter 0.4.5 (Hex package) (mix)
  locked at 0.4.5 (websock_adapter) 1d9812dc
  ok
```

* Operating system
  * macOS Big Sur Version 11.6

### Expected behavior
Working instructions for how to get a phoenix server up and running.
I am currently using https://hexdocs.pm/phoenix/up_and_running.html

```
brew install postgresql
brew install elixir
mix local.hex
mix archive.install hex phx_new
mix phx.new hello
cd hello
mix ecto.create
mix phx.server
```

### Actual behavior
When attempting to launch phoenix server via "mix phx.server",
I get an error indicating an invalid certificate authority for tailwind.
Regardless of whatever the problem happens to be with tailwind,
the effect is that it is impossible to create even the most simple hello world application with the phoenix framework, 
so it is the phoenix project generator that needs to be fixed,
possibly by finding an alternative to the version of tailwind that is causing the ultimate problem. 

```shell
[info] Running HelloWeb.Endpoint with cowboy 2.9.0 at 127.0.0.1:4000 (http)
[debug] Downloading tailwind from https://github.com/tailwindlabs/tailwindcss/releases/download/v3.2.4/tailwindcss-macos-x64
[info] Access HelloWeb.Endpoint at http://localhost:4000
[watch] build finished, watching for changes...
[notice] TLS :client: In state :wait_cert_cr at ssl_handshake.erl:2111 generated CLIENT ALERT: Fatal - Unknown CA

[error] Task #PID<0.586.0> started from HelloWeb.Endpoint terminating
** (RuntimeError) couldn't fetch https://github.com/tailwindlabs/tailwindcss/releases/download/v3.2.4/tailwindcss-macos-x64: {:error, {:failed_connect, [{:to_address, {'github.com', 443}}, {:inet, [:inet], {:tls_alert, {:unknown_ca, 'TLS client: In state wait_cert_cr at ssl_handshake.erl:2111 generated CLIENT ALERT: Fatal - Unknown CA\n'}}}]}}
(tailwind 0.1.10) lib/tailwind.ex:348: Tailwind.fetch_body!/1
(tailwind 0.1.10) lib/tailwind.ex:235: Tailwind.install/1
(tailwind 0.1.10) lib/tailwind.ex:215: Tailwind.install_and_run/2
(phoenix 1.7.1) lib/phoenix/endpoint/watcher.ex:19: Phoenix.Endpoint.Watcher.watch/2
(elixir 1.14.3) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
(stdlib 4.3) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
Function: &Phoenix.Endpoint.Watcher.watch/2
Args: ["tailwind", {Tailwind, :install_and_run, [:default, ["--watch"]]}]
```

**Comment by SeanShubin** [comment]
Mentioned this with the tailwind developers as well:
https://github.com/tailwindlabs/tailwindcss/discussions/10589#discussioncomment-5271349

As it stands, it really harms anyone's ability to get started with Phoenix Framework, as anyone trying to evaluate if Phoenix Framework is a good solution or not gets immediately shut down because they can't follow the instructions to get the simple hello application running, so are more likely to go with something like react that just works the first try.

**Comment by josevalim** [comment]
@SeanShubin can you consistently reproduce the error? Either GitHub temporarily refused the connection (and it should work on the next try) or are you behind an http_proxy?

**Comment by josevalim** [comment]
If you can’t download it with cURL it seems something else is up. Are you behind an http proxy by any chance?

**Comment by puruzio** [comment]
Hi Jose, While we are waiting for @SeanShubin to respond,  I'm having the same issue on Windows behind corporate firewall.  Is there a solution in that case? I can download the file manually but don't know how to make mix/Tailwind use it instead of trying to download the file itself.

**Comment by josevalim** [comment]
@puruzio can you please try tailwind from github? it is allegedly fixed there. :) and we can ship a new release soon. Thank you for following up!

**Comment by josevalim** [comment]
To be clear, something like this in your mix.exs:

    {:tailwind, github: "phoenixframework/tailwind"}

:)

**Comment by puruzio** [comment]
Thank you.  `mix deps.get` ran fine but then I got the following after `mix phx.server`

```

17:27:48.774 [error] Task #PID<0.1761.0> started from MyAppWeb.Endpoint terminating
** (RuntimeError) couldn't fetch https://github.com/tailwindlabs/tailwindcss/releases/download/v3.2.7/tailwindcss-windows-x64.exe: {:error, {:failed_connect, [{:to_address, {'objects.githubusercontent.com', 443}}, {:inet, [:inet], {:tls_alert, {:unknown_ca, 'TLS client: In state certify at ssl_handshake.erl:2098 generated CLIENT ALERT: Fatal - Unknown CA\n'}}}]}}
    (tailwind 0.2.0-dev) lib/tailwind.ex:346: Tailwind.fetch_body!/1
    (tailwind 0.2.0-dev) lib/tailwind.ex:235: Tailwind.install/1
    (tailwind 0.2.0-dev) lib/tailwind.ex:215: Tailwind.install_and_run/2
    (phoenix 1.7.1) lib/phoenix/endpoint/watcher.ex:19: Phoenix.Endpoint.Watcher.watch/2
    (elixir 1.14.0) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (stdlib 4.1) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
Function: &Phoenix.Endpoint.Watcher.watch/2
    Args: ["tailwind", {Tailwind, :install_and_run, [:default, ["--watch"]]}]

17:27:48.776 [debug] Downloading tailwind from https://github.com/tailwindlabs/tailwindcss/releases/download/v3.2.7/tailwindcss-windows-x64.exe

17:27:49.295 [notice] TLS :client: In state :certify at ssl_handshake.erl:2098 generated CLIENT ALERT: Fatal - Unknown CA
```

**Comment by josevalim** [comment]
Sorry to hear. It seems there is something still going with the proxy. I will improve the error message to include steps to do it manually, meanwhile, you can use npm and point to it yourself:

https://github.com/phoenixframework/tailwind/blob/ee5a33c0aa981f77f99ae1e1d316ee643595eb41/lib/tailwind.ex#L25

**Comment by josevalim** [comment]
You can also download it yourself and put it at:

    _build/tailwindcss-windows-x64.exe

**Comment by puruzio** [comment]
> You can also download it yourself and put it at:
> 
> ```
> _build/tailwind-windows-x64.exe
> ```

This worked like a charm, coupled with this in config.exs.
 ```
watchers: [
 ...
    tailwind: {Tailwind, :run, [:default, ~w(--watch)]}
  ],
```

Thank you, Jose!

**Comment by SeanShubin** [comment]
Thanks everybody for looking into it.  Your analysis was very helpful.  Here is what I ended up doing

```shell
curl -sLO https://github.com/tailwindlabs/tailwindcss/releases/download/v3.3.2/tailwindcss-macos-x64
chmod +x tailwindcss-macos-x64
mv tailwindcss-macos-x64 _build/tailwind-macos-x64
```

I actually confirmed this empirically by alternately running with my company vpn turned on and off, until it eventually just started working.  Then I did a diff on the directory structure to see exactly what changed from not-working to working.  The only change was that file named tailwind-macos-x64 in the _build directory.

This is pretty close to some of the other suggestions, with a different name.

It is weird that it doesn't work with vpn, and it doesn't work without vpn, but if I keep alternating it eventually works.   I have a workaround for now, but I am hoping this gets fixed in the future.  It is odd that tailwind is the ONLY dependency that gives me this problem.  Thanks again everyone for your help.
